RevenueCatUI/Templates/Example1Template.swift

```diff
@@ -35,21 +41,37 @@ struct Example1Template: TemplateViewType {
     var body: some View {
         switch self.data {
         case let .success(data):
-            Example1TemplateContent(data: data)
+            Example1TemplateContent(data: data, introEligibility: self.introEligibility)
+                .task(id: self.package) {
+                    if let package = self.package {
+                        self.introEligibility = await self.introEligibilityChecker.eligibility(
+                            for: package.storeProduct
```

Comment:
might as well move this logic into the intro eligibility checker so we can do 
```swift
self.introEligibility = await self.introEligibilityChecker.eligibility(for: self.package)
```

Reply:
Turned that into an extension.